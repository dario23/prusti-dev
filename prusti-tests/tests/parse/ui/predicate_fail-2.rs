// compile-flags: -Pprint_desugared_specs=true -Pprint_typeckd_specs=true -Pno_verify=true -Phide_uuids=true
// normalize-stdout-test: "[a-z0-9]{32}" -> "$(NUM_UUID)"
// normalize-stdout-test: "[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}" -> "$(UUID)"

/// Failure case for predicate parsing: can only be used on function definitions

use prusti_contracts::*;

// doesn't work on just function decl
#[predicate]
fn result_is_one() -> bool;

// doesn't work on non-function-y items
#[predicate]
static FOO: usize = 0;

fn main() {}
