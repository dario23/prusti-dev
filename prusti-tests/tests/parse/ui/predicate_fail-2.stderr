error: macros that expand to items must be delimited with braces or followed by a semicolon
  --> $DIR/predicate_fail-2.rs:11:1
   |
11 | fn result_is_one() -> bool;
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
help: change the delimiters to curly braces
   |
11 | {}
   | ^
help: add a semicolon
   |
11 | fn result_is_one() -> bool;;
   |                            ^

error: you can only use `#[predicate]` on function definitions
  --> $DIR/predicate_fail-2.rs:11:1
   |
11 | fn result_is_one() -> bool;
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: macros that expand to items must be delimited with braces or followed by a semicolon
  --> $DIR/predicate_fail-2.rs:15:1
   |
15 | static FOO: usize = 0;
   | ^^^^^^^^^^^^^^^^^^^^^^
   |
help: change the delimiters to curly braces
   |
15 | {}
   | ^
help: add a semicolon
   |
15 | static FOO: usize = 0;;
   |                       ^

error: you can only use `#[predicate]` on function definitions
  --> $DIR/predicate_fail-2.rs:15:1
   |
15 | static FOO: usize = 0;
   | ^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 4 previous errors

